src/background/quality/providers/QualityProvider.test.ts

```diff
@@ -0,0 +1,144 @@
+import { Quality } from "../../../types/Quality";
+import { QualityProvider } from "./QualityProvider";
+
+let browser: Browser;
+let mockBrowser: MockzillaDeep<Browser>;
+let qualityProvider: QualityProvider;
+
+beforeEach(() => {
+  [browser, mockBrowser] = deepMock<Browser>("browser", false);
+  mockBrowser.storage.local.mockAllow();
+
+  qualityProvider = new QualityProvider(browser.storage.local);
+});
+
+test("strips everything off url, except from domain", () => {
+  expect(
+    qualityProvider["stripUrl"]("https://www.example.com/search?key=example")
+  ).toEqual("www.example.com");
+});
+
+test("returns quality for url, if domain is cached", async () => {
+  mockBrowser.storage.local.get
+    .expect(["qualities"])
+    .andResolve({ qualities: { "www.example.com": Quality.DomainValidated } });
+
+  await expect(
+    qualityProvider["getQuality"]("https://www.example.com")
+  ).resolves.toEqual(Quality.DomainValidated);
+});
+
+test("returns unknown quality for url, if domain is not cached", async () => {
+  mockBrowser.storage.local.get.expect(["qualities"]).andResolve({});
+
+  await expect(
+    qualityProvider["getQuality"]("https://www.example.com")
+  ).resolves.toEqual(Quality.Unknown);
+});
+
+test("throws StorageError, if storage get operation fails", async () => {
+  mockBrowser.storage.local.get.expect(["qualities"]).andReject(new Error());
+
+  await expect(
+    qualityProvider["getQuality"]("https://www.example.com")
+  ).rejects.toBeInstanceOf(StorageError);
+});
+
+test("sets quality for provided url, if quality is stored already", async () => {
+  mockBrowser.storage.local.get.expect(["qualities"]).andResolve({
+    qualities: { "www.example.com": Quality.OrganizationValidated },
+  });
+  mockBrowser.storage.local.set
+    .expect({ qualities: { "www.example.com": Quality.DomainValidated } })
+    .andResolve();
+
+  await expect(
+    qualityProvider.setQuality(
+      "https://www.example.com",
+      Quality.DomainValidated
+    )
+  ).resolves.toEqual(undefined);
+});
+
+test("sets quality for provided url, if quality is not stored already", async () => {
+  mockBrowser.storage.local.get.expect(["qualities"]).andResolve({});
+  mockBrowser.storage.local.set
+    .expect({ qualities: { "www.example.com": Quality.DomainValidated } })
+    .andResolve();
+
+  await expect(
+    qualityProvider.setQuality(
+      "https://www.example.com",
+      Quality.DomainValidated
+    )
+  ).resolves.toEqual(undefined);
+});
+
+test("throws Storage, if storage set operation fails", async () => {
+  mockBrowser.storage.local.get.expect(["qualities"]).andResolve({});
+  mockBrowser.storage.local.set
+    .expect({ qualities: { "www.example.com": Quality.DomainValidated } })
+    .andReject(new Error());
+
+  await expect(
+    qualityProvider.setQuality(
+      "https://www.example.com",
+      Quality.DomainValidated
+    )
+  ).rejects.toBeInstanceOf(StorageError);

```

Comment:
Should this storage error be used in the logging persistence manager too?

Reply:
It represents a failed storage operation in the web extension storage, so yes if you want to.